load_chinese_lattice.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# NOTE: this script is based on load_clancy_lattice.py to do a basic import of Chinese dictionary data
+import re
+import time
+from typing import Optional, Tuple
+
+from lattices.models import LatticeNode, LemmaNode
+
+
+DICT_FILE = "import-data/cedict_ts.u8"
+DEFAULT_LEMMA_NODE_CONTEXT = "chinese"
+
+
+def create_lemma_node(lemma, lattice_node, context=DEFAULT_LEMMA_NODE_CONTEXT):
+    lemma_node, created = LemmaNode.objects.get_or_create(
+        context=context,
+        lemma=lemma,
+        defaults={
+            "node": lattice_node,
+        }
+    )
+    if created:
+        print("  created", context, "lemma node", lemma_node.pk, lemma)
+    else:
+        existing_lattice_node = lemma_node.node
+        print(" ", context, "node already existed pointing to lattice node", existing_lattice_node.pk, existing_lattice_node.label)
+        if existing_lattice_node.canonical:
+            parent_lattice_node = LatticeNode.objects.create(label=lemma, gloss="from " + context, canonical=False)
+            parent_lattice_node.children.add(existing_lattice_node)
+            parent_lattice_node.children.add(lattice_node)
+            parent_lattice_node.save()
+            lemma_node.node = parent_lattice_node
+            lemma_node.save()
+            print("  created parent lattice node", parent_lattice_node.pk, parent_lattice_node.label)
+            print("  lattice node", existing_lattice_node.pk, existing_lattice_node.label, "put under", parent_lattice_node.pk, parent_lattice_node.label)
+            print("  lattice node", lattice_node.pk, lattice_node.label, "put under", parent_lattice_node.pk, parent_lattice_node.label)
+            print(" ", context, "node now points to lattice node", lemma_node.node.pk, lemma_node.node.label)
+        else:
+            existing_lattice_node.children.add(lattice_node)
+            print("  lattice node", lattice_node.pk, lattice_node.label, "put under", existing_lattice_node.pk, existing_lattice_node.label)
+            existing_lattice_node.save()
+
+
+def get_or_create_lattice_node(label: str, gloss) -> Tuple[Optional[LemmaNode], bool]:
+    return LatticeNode.objects.get_or_create(label=label, gloss=gloss, canonical=False)
+
+
+def node_exists(lemma: str) -> bool:
+    return LemmaNode.objects.filter(lemma=lemma, context=DEFAULT_LEMMA_NODE_CONTEXT).exists()
+
+
+row_grouping_re = re.compile(r"^(?P<lemma>\S+) (?P<altform>\S+) (?P<defn>.*)$")
+
+ROW_COUNT_FOR_UPDATE = 100
+ROW_TO_START_FROM = 98500
+current_row = 0
+start_time = time.time()
+block_start_time = start_time
+rows = []
+
+with open(DICT_FILE) as f:
+    rows = f.readlines()
```

Comment:
This was to see if loading everything into memory would improve performance -- it wasn't the bottleneck, though.